Add The Unlicense to GPL-compatible license check

john-shaffer opened the pull request:
The Unlicense is a GPL-Compatible free software
license, but is not recognized by the is_license_gpl_compatible check.
See https://www.gnu.org/licenses/license-list.html#Unlicense.

- Add The Unlicense to the regex in is_license_gpl_compatible() so the plugin will correctly recognize it as GPL-compatible.
- Add tests for recognition of The Unlicense

john-shaffer committed:
Add The Unlicense to GPL-compatible license check

The Unlicense is a GPL-Compatible free software
license
See https://www.gnu.org/licenses/license-list.html#Unlicense.

- Add The Unlicense to the regex in
  is_license_gpl_compatible() so the plugin will
  correctly recognize it as GPL-compatible.
- Add tests for recognition of The Unlicense

ernilambar commented:
@john-shaffer Thanks for the PR. I believe following methods also need to be updated in `License_Utils.php`
- get_normalized_license()
- is_license_valid_identifier()

You can test it by keeping `License: Unlicense` in plugin header.

davidperezgar commented:
@john-shaffer did you see the comments?

john-shaffer committed:
Normalize The Unlicense to SPDX identifier

According to https://spdx.org/licenses/
the full name is "The Unlicense" and the SPDX
identifier is "Unlicense".

Update get_normalized_license to transform
both "The Unlicense" and "Unlicense",
case-insensitive, to "Unlicense".

This allows "The Unlicense" to be recognized in
a plugin's readme.txt. It also fixes a case where
if a plugin file said "The Unlicense" but the
readme.txt said "Unlicense", Plugin Check would
report that there was a license mismatch.

ernilambar reviewed (approved):
LGTM.
